src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_device_state_changed_notification.cc

```diff
@@ -38,40 +38,21 @@
 #include "encryption/hashing.h"
 
 namespace {
-// TODO(AOleynik) : replace this !!!
-void check_zero(const char& a, const char& b, std::string& bt_mac) {
-  if ('0' != a && '0' == b) {
-    bt_mac.push_back(a);
-    bt_mac.push_back(b);
-  } else if ('0' == a) {
-    bt_mac.push_back(b);
-  } else {
-    bt_mac.push_back(a);
-    bt_mac.push_back(b);
-  }
-}
-
-std::string convert_to_bt_mac(std::string& deviceInternalId) {
-  std::transform(deviceInternalId.begin(),
-                 deviceInternalId.end(),
-                 deviceInternalId.begin(),
-                 ::tolower);
-
-  std::string bt_mac;
-  check_zero(deviceInternalId[10], deviceInternalId[11], bt_mac);
-  bt_mac.push_back(':');
-  check_zero(deviceInternalId[8], deviceInternalId[9], bt_mac);
-  bt_mac.push_back(':');
-  check_zero(deviceInternalId[6], deviceInternalId[7], bt_mac);
-  bt_mac.push_back(':');
-  check_zero(deviceInternalId[4], deviceInternalId[5], bt_mac);
-  bt_mac.push_back(':');
-  check_zero(deviceInternalId[2], deviceInternalId[3], bt_mac);
-  bt_mac.push_back(':');
-  check_zero(deviceInternalId[0], deviceInternalId[1], bt_mac);
 
-  return bt_mac;
+std::string convert_to_bt_mac(const std::string& device_internal_id) {
+  std::stringstream bt_mac;
+  for (size_t i = 0; i < device_internal_id.length(); i += 2) {
```

Comment:
@ValeriiMalkov what mean magic 2?

Reply:
@KVGrygoriev this magic number means the length of the mac address octet when it is represented as a hex string. Here we have to process every two hex symbols of the one octet per loop iteration. I moved this magic number to a named constant to make this code cleaner.
I've also added a missed description for this function. Please check for updates in https://github.com/smartdevicelink/sdl_core/pull/2668/commits/254ab750b4ea2f392dfd9f321b8e322de68212d9

Reply:
@AKalinich-Luxoft , thanks. Good explanation and code is clean as it should be